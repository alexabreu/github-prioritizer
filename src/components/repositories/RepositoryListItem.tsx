import React, { Component, MouseEvent } from 'react';
import { withRouter, RouteComponentProps } from 'react-router';

import { ListItem, ListItemText, ListItemAvatar, Avatar } from '@material-ui/core';



export interface RepositoryListItemProps {
  repository: Repository;
  onSelect: (id: Repository['id'], repositoryUrl: Repository['url']) => void;
  selectedRepository: Repository | undefined;
}

class RepositoryListItem extends Component<RepositoryListItemProps & RouteComponentProps> {
  public render() {
    const { repository, selectedRepository } = this.props;

    return (
      <ListItem
        alignItems="flex-start"
        button onClick={this.handleSelect}
        selected={repository.id === (selectedRepository && selectedRepository.id)}
      >
        <ListItemAvatar>
          <Avatar alt={repository.owner.login} title={repository.owner.login} src={repository.owner.avatar_url} />
        </ListItemAvatar>
        <ListItemText
          primary={repository.name}
          secondary={repository.description}
        />
      </ListItem>
    );
  }

  private handleSelect = (e: MouseEvent<HTMLElement>) => {
    const { onSelect, repository, history } = this.props;

    history.replace(`/repos/${repository.id}/issues`);
    onSelect(repository.id, repository.url);
  }
}

export default withRouter(RepositoryListItem);
