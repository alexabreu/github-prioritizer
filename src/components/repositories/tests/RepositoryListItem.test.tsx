import React from 'react';
import { createRender } from '@material-ui/core/test-utils';

import RepositoryListItem from '../RepositoryListItem';
import { MemoryRouter } from 'react-router';

const repository: Repository = {
  id: 456,
  full_name: 'Repo repo man',
  name: 'repo man',
  url: 'https://awesome.com',
  description: 'stuff stuf stuff',
  private: false,
  owner: {
    login: "dude #1",
    id: 8910,
    avatar_url: 'https://picture1.com',
    url: 'https://dude1.com',

  }
};

describe('<RepositoryListItem/>', () => {
  test('it renders correctly', () => {
    const render = createRender();
    expect(render(
      <MemoryRouter>
        <RepositoryListItem
          selectedRepository={repository}
          repository={repository}
          isLoadingIssues={false}
          onSelect={() => null}
        />
      </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
