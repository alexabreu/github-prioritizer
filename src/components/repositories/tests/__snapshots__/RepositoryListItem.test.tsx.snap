// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryListItem/> it renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiButtonBase-root-13 MuiListItem-root-1 MuiListItem-default-4 MuiListItem-dense-5 MuiListItem-gutters-9 MuiListItem-button-10 MuiListItem-alignItemsFlexStart-6 MuiListItem-selected-12",
      "role": "button",
      "tabindex": "0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiAvatar-root-19 MuiListItemAvatar-alignItemsFlexStart-17",
          "title": "dude #1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "dude #1",
              "class": "MuiAvatar-img-21",
              "src": "https://picture1.com",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiListItemText-root-22",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25",
              },
              "children": Array [
                Object {
                  "data": "repo man",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26",
                },
                "children": Array [
                  Object {
                    "data": "stuff stuf stuff",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26",
              },
              "children": Array [
                Object {
                  "data": "stuff stuf stuff",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25",
                },
                "children": Array [
                  Object {
                    "data": "repo man",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "title": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiListItemText-root-22",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25",
            },
            "children": Array [
              Object {
                "data": "repo man",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26",
              },
              "children": Array [
                Object {
                  "data": "stuff stuf stuff",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-61 MuiListItemText-secondary-26",
            },
            "children": Array [
              Object {
                "data": "stuff stuf stuff",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25",
              },
              "children": Array [
                Object {
                  "data": "repo man",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiAvatar-root-19 MuiListItemAvatar-alignItemsFlexStart-17",
            "title": "dude #1",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "dude #1",
                "class": "MuiAvatar-img-21",
                "src": "https://picture1.com",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "title": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
